### PR TITLE
feat(doc-framework): Allow for 'next' core component usage alongside core components

### PR DESCRIPTION
**Pull request by jeffpuzzo**
**Closes**: #3125
**Depends on**: https://github.com/patternfly/patternfly-react/pull/7824

- Give "next" module components unique key names when building prop components.
- Filter out duplicate component names when building "scope" for the list of examples we render on any given documentation page (based on the "source" passed from react-docs).

### CODE REVIEW

File: packages/documentation-framework/components/example/example.js [hunk 1]
```diff
@@ -98,12 +97,22 @@ export const Example = ({
   const [editorCode, setEditorCode] = React.useState(code);
   const loc = useLocation();
 
+  // Build unique list of module components for when core module names overlap with "next" module names.
+  const moduleComponents = Object.entries({ ...reactCoreModule, ...reactTableModule }).reduce((acc, [componentName, componentValue]) => {
```
Comment (jschuler) [inline packages/documentation-framework/components/example/example.js:101]: could this more simply be
```
const moduleComponents = {
   ...reactCoreModule,
   ...reactTableModule,
   ...(source === 'react-next' ? reactCoreNextModule : {})
}
```

File: packages/documentation-framework/scripts/md/parseMD.js [hunk 2]
```diff
@@ -60,15 +60,17 @@ function toReactComponent(mdFilePath, source, buildMode) {
...
-            return true;
-          }
-          file.message(`Prop component ${propComponent} missing from tsDocgen`);
-          return false;
-        })
-        .map(propComponent => tsDocs[propComponent])
+      const propComponents = [...new Set(frontmatter.propComponents || [])].reduce((acc, componentName) => {
+        const name = getTsDocName(componentName, source === 'react-next');
+
+        if (tsDocs[name]) {
```
Comment (jschuler) [inline packages/documentation-framework/scripts/md/parseMD.js:66]: Trying to understand. So if there's let's say ...next/components/Wizard/Wizard.tsx, then it exists as `tsDocs['Wizard-next']`?
And then it will get pushed to 
```
propComponents[
  {
    "name": "Wizard-next",
    "props": []
  }
]
```
?

File: packages/documentation-framework/scripts/md/typecheck.js [hunk 3]
```diff
@@ -34,13 +34,18 @@ declare module '\\*.svg' {
 const reactStylesDir = path.join(require.resolve('@patternfly/react-styles/package.json'), '../');
 const reactStyles = glob.sync(path.join(reactStylesDir, 'css/**/*.d.ts'))
   .map(f => f.replace(reactStylesDir, '@patternfly/react-styles/').replace(/\.d.ts$/, ''));
+const defaultImports = [
+  'react',
+  '@reach/router',
+  '@patternfly/react-core/next',
```
Comment (jschuler) [inline packages/documentation-framework/scripts/md/typecheck.js:40]: is this not added to versions.json?

File: packages/documentation-framework/scripts/webpack/webpack.base.config.js [hunk 4]
```diff
@@ -117,6 +117,7 @@ module.exports = (_env, argv) => {
         'client-styles': path.resolve(process.cwd(), 'patternfly-docs/patternfly-docs.css.js'),
         './routes-client': path.resolve(process.cwd(), 'patternfly-docs/patternfly-docs.routes.js'),
         './routes-generated': path.resolve(process.cwd(), 'patternfly-docs/generated/index.js'),
+        '@patternfly/react-core/next': '@patternfly/react-core/dist/js/next/index.js',
```
Comment (jschuler) [inline packages/documentation-framework/scripts/webpack/webpack.base.config.js:120]: why is this needed?

Comment (jeffpuzzo) [reply packages/documentation-framework/scripts/webpack/webpack.base.config.js:120]: It shouldn't be after the next path exists in patternfly-react. This PR wasn't quite ready to be reviewed as it depends on the version of react-core that has the "next" path.

File: packages/documentation-framework/scripts/md/parseMD.js (hunk 2 above)
Comment (jeffpuzzo) [reply packages/documentation-framework/scripts/md/parseMD.js:66]: No, the name of the component would be "Wizard" still, but the props would be that of the appropriate component.

Here's an example generated output for the "Next" Wizard:
```
const pageData = {
  "id": "Wizard",
  "section": "components",
  "source": "react-next",
  "slug": "/components/wizard/react-next",
  "sourceLink": "https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/next/components/Wizard/examples/Wizard.md",
  "propComponents": [
    {
      "name": "Wizard",
```

File: packages/documentation-framework/scripts/md/typecheck.js (hunk 3 above)
Comment (jeffpuzzo) [reply packages/documentation-framework/scripts/md/typecheck.js:40]: I did not update that file. What are the expectations there and what is it used for exactly?

File: packages/documentation-framework/scripts/webpack/webpack.base.config.js (hunk 4 above)
Comment (jeffpuzzo) [reply packages/documentation-framework/scripts/webpack/webpack.base.config.js]: Spoke with @jschuler in slack about this and we decided to keep this around until webpack is updated to v5 for the documentation-framework.

**Review by tlabaj — DISMISSED** [review]
I am not familiar with the framework.  Looks good to me if @jschuler or @CooperRedhat are good with it.

File: packages/documentation-framework/components/example/example.js [hunk 5]
```diff
@@ -77,8 +78,6 @@ export const Example = ({
   liveContext,
   // Content that appears between h3 and code block to explain example
   children,
-  // Show dark theme switcher on full page examples
-  hasDarkThemeSwitcher = process.env.hasDarkThemeSwitcher
```
Comment (nicolethoen) [inline packages/documentation-framework/components/example/example.js:81]: Why did you remove this? I think we want the dark theme switcher on full pages if the environment sets `hasDarkThemeSwitcher` to true.

And it is also seeming like the full page demos are not working in the review? 
https://patternfly-org-pr-3126-v4.surge.sh/v4/components/notification-drawer/react-demos/basic/
https://patternfly-org-pr-3126-v4.surge.sh/v4/components/page/react-demos/sticky-section-group-using-pageheader/

Comment (jeffpuzzo) [reply packages/documentation-framework/components/example/example.js:81]: Ah, not intentional. Will add this back.

Comment (jeffpuzzo) [reply packages/documentation-framework/components/example/example.js]: Both of those links point to their respective pages now fine it appears.

**Review by jschuler — APPROVED** [review]
